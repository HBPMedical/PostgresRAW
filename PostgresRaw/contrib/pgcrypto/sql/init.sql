--
-- init pgcrypto
--

--
-- first, define the functions.  Turn off echoing so that expected file
-- does not depend on contents of pgcrypto.sql.
--
SET client_min_messages = warning;
\set ECHO none
\i pgcrypto.sql
\set ECHO all
RESET client_min_messages;

-- ensure consistent test output regardless of the default bytea format
SET bytea_output TO escape;

-- check for encoding fn's
SELECT encode('foo', 'hex');
SELECT decode('666f6f', 'hex');

-- check error handling
select gen_salt('foo');
select digest('foo', 'foo');
select hmac('foo', 'foo', 'foo');
select encrypt('foo', 'foo', 'foo');

