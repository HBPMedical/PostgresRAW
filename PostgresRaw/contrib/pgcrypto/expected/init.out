--
-- init pgcrypto
--
--
-- first, define the functions.  Turn off echoing so that expected file
-- does not depend on contents of pgcrypto.sql.
--
SET client_min_messages = warning;
\set ECHO none
RESET client_min_messages;
-- ensure consistent test output regardless of the default bytea format
SET bytea_output TO escape;
-- check for encoding fn's
SELECT encode('foo', 'hex');
 encode 
--------
 666f6f
(1 row)

SELECT decode('666f6f', 'hex');
 decode 
--------
 foo
(1 row)

-- check error handling
select gen_salt('foo');
ERROR:  gen_salt: Unknown salt algorithm
select digest('foo', 'foo');
ERROR:  Cannot use "foo": No such hash algorithm
select hmac('foo', 'foo', 'foo');
ERROR:  Cannot use "foo": No such hash algorithm
select encrypt('foo', 'foo', 'foo');
ERROR:  Cannot use "foo": No such cipher algorithm
